microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import Path
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import Gpu, SerialConsole
+from lisa.features.gpu import ComputeSDK
+from lisa.operating_system import Redhat
+from lisa.sut_orchestrator.azure.common import AZURE, AzureNodeSchema
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lsmod
+from lisa.util import LisaException, SkippedException
+from lisa.util.perf_timer import create_timer
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+    tags=[],
+)
+class gpu(TestSuite):
+    grid_skus = ["Standard_NV"]
+    cuda_skus = ["Standard_NC", "Standard_ND"]
+
+    def __init__(self, node: Node) -> None:
+        self.gpu_feature = node.features[Gpu]
```

Comment:
Avoid to execute long time logic in `__init__`. Those feature, tool should be gotten in test case, not in test suite level. It may causes issue, if the node is different for two test cases.

Reply:
Hmm so should import these features and tools in each test case? I am not sure what benefit we are getting then having more than one test be part of a single test-suite?
Shouldn't all the tests of a test-suite run on the same node?

Reply:
The test suite is used to share logic, not runtime data. Because each test may have different priority and requirement, they may run in different batches or nodes.